templates/test/dev/patches/control-plane-custom-builds.yaml

```diff
@@ -25,13 +25,24 @@ spec:
         set -o errexit
 
         systemctl stop kubelet
-        declare -a BINARIES=("kubeadm" "kubectl" "kubelet")
-        for BINARY in "$${BINARIES[@]}"; do
-          echo "* installing package: $${BINARY} ${KUBE_GIT_VERSION}"
-          curl --retry 10 --retry-delay 5 -w "response status code is %{http_code}" "https://${AZURE_STORAGE_ACCOUNT}.blob.core.windows.net/${AZURE_BLOB_CONTAINER_NAME}/${KUBE_GIT_VERSION}/bin/linux/amd64/$${BINARY}" --output "/usr/bin/$${BINARY}"
-        done
+        # Run the az login command with managed identity
+        if az login --identity > /dev/null 2>&1; then
+          echo "Logged in Azure with managed identity"
+          declare -a BINARIES=("kubeadm" "kubectl" "kubelet")
+          for BINARY in "$${BINARIES[@]}"; do
+            echo "* installing package: $${BINARY} ${KUBE_GIT_VERSION}"
+            az storage blob download --blob-url "https://${AZURE_STORAGE_ACCOUNT}.blob.core.windows.net/${AZURE_BLOB_CONTAINER_NAME}/${KUBE_GIT_VERSION}/bin/linux/amd64/$${BINARY}" -f "/usr/bin/$${BINARY}" --auth-mode login
+          done
+        else
+          echo "Using curl to download the binaries"
+          declare -a BINARIES=("kubeadm" "kubectl" "kubelet")
```

Comment:
nit: can this and line 31 be moved out of the if/else? Not sure how `declare` works though so it's not a problem if it changes behavior.

Reply:
Good catch, pushing the changes..

Reply:
Pushed the changes. Please take a look!